File: Algorithms/p406_Queue_Reconstruction_by_Height_2.py
class Solution(object):
    def reconstructQueue(self, people):
        """
        :type people: List[List[int]]
        :rtype: List[List[int]]
        """
        res = []
        for h,k in sorted(people,key=lambda (h,k):(-h,k)):
            res.insert(k,[h,k])
        return res

File: Algorithms/p406_Queue_Reconstruction_by_Height/p406_Queue_Reconstruction_by_Height_2.cpp
class Solution {
public:
    vector<pair<int, int>> reconstructQueue(vector<pair<int, int>>& people) {
        sort(people.begin(),people.end(),[](pair<int,int>& a, pair<int,int>& b){ return a.first>b.first || a.first==b.first&&a.second<b.second; });
        vector<pair<int, int>> res;
        for(auto &e : people){
            res.insert(res.begin()+e.second,e);
        }
        return res;
    }
};
